src/ray/raylet/node_manager.cc

```diff
@@ -1169,6 +1169,10 @@ void NodeManager::ProcessDisconnectClientMessage(
 
     // Since some resources may have been released, we can try to dispatch more tasks.
     DispatchTasks(local_queues_.GetReadyTasksByClass());
+
+    // Remove the worker's lease. If the lessee tries to return it, it will get
+    // an error status.
+    leased_workers_.erase(worker->Port());
```

Comment:
I do it in a slightly different place here: https://github.com/ray-project/ray/pull/6177/files#diff-50e88810cd198c83bbce3317b1bc9924R1119

I guess both are equivalent?

Reply:
Hmm okay that seems fine. I don't really understand that code, but yeah, they seem equivalent.